src/sentry/utils/snuba.py

```diff
@@ -1565,7 +1572,7 @@ def get_related_project_ids(column, ids):
     return []
 
 
-def shrink_time_window(issues, start):
+def shrink_time_window(issues, start: datetime) -> datetime:
```

Comment:
Doesn't `issues` need a type annotation?

Reply:
@evanh The pyproject.toml in sentry repo doesn't actually require type annotations for function signatures. I think its only to the level of verifying consistent typing where they are defined. I think might be able to opt-in to stricter type checking that would require fully typed function signatures, but there are quite a bit of untyped functions I think.

Reply:
ideally it should have one -- or the function isn't getting full checking.  at the moment we're in the gradual phase of trying to remove everything from the blocklist so partial typing is fine.  in the future we'll tighten this to require full signatures probably